swan/files/swan_config.py

```diff
@@ -24,10 +24,14 @@ def __init__(self, spawner, pod):
 
     def get_swan_user_pod(self):
 
+        safe_username = escapism.escape(
+            self.spawner.user.name, safe = self.spawner.safe_chars, escape_char = '-'
+        ).lower()
+
         # pod labels
         pod_labels = dict(
             lcg_release = self.spawner.user_options[self.spawner.lcg_rel_field].split('/')[0],
-            swan_user = self.spawner.user.name
+            swan_user = safe_username # FIXME why do we need a swan_user if we have username already?
```

Comment:
I guess it is not strictly needed since the pod name already contains the user name, it is just a label that is added to the user pod to add some extra metadata. Perhaps if a user had multiple servers it would be useful to filter only those of the users. I guess it doesn't hurt to keep it.

But why do you need to escape the user name before using it in a label?

Reply:
Because it can contain invalid chars. I.e, external accounts in our SSO have the username in the format "some_id@service_name".
It's good practice.

But for the main thing: we already have the username as label (`hub.jupyter.org/username`), that's why I asked.